internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -387,20 +381,188 @@ func resourceKustoClusterCreateUpdate(d *pluginsdk.ResourceData, meta interface{
+		expandedIdentity, err := identity.ExpandSystemAndUserAssignedMap(d.Get("identity").([]interface{}))
+		if err != nil {
+			return fmt.Errorf("expanding `identity`: %+v", err)
+		}
+		model.Identity = expandedIdentity
+	}
+
+	if d.HasChange("allowed_fqdns") {
+		props.AllowedFqdnList, err = expandKustoListString(d.Get("allowed_fqdns").([]interface{}))
+		if err != nil {
+			return err
+		}
+	}
+
+	if d.HasChange("allowed_ip_ranges") {
+		props.AllowedIPRangeList, err = expandKustoListString(d.Get("allowed_ip_ranges").([]interface{}))
+		if err != nil {
+			return err
+		}
+	}
+
+	if d.HasChange("auto_stop_enabled") {
+		props.EnableAutoStop = utils.Bool(d.Get("auto_stop_enabled").(bool))
+	}
+
+	if d.HasChange("disk_encryption_enabled") {
+		props.EnableDiskEncryption = utils.Bool(d.Get("disk_encryption_enabled").(bool))
+	}
+
+	if d.HasChange("double_encryption_enabled") {
+		props.EnableDoubleEncryption = utils.Bool(d.Get("double_encryption_enabled").(bool))
+	}
+
+	if d.HasChange("language_extensions") {
+		if features.FourPointOhBeta() {
+			if v, ok := d.GetOk("language_extensions"); ok {
+				props.LanguageExtensions = expandKustoClusterLanguageExtensionList(v.([]interface{}))
+			}
+		} else {
+			props.LanguageExtensions = expandKustoClusterLanguageExtensions(d)
+		}
+	}
+
+	if d.HasChange("outbound_network_access_restricted") {
+		restrictOutboundNetworkAccess := clusters.ClusterNetworkAccessFlagDisabled
+		if d.Get("outbound_network_access_restricted").(bool) {
+			restrictOutboundNetworkAccess = clusters.ClusterNetworkAccessFlagEnabled
+		}
+		props.RestrictOutboundNetworkAccess = pointer.To(restrictOutboundNetworkAccess)
+	}
+
+	if d.HasChange("purge_enabled") {
+		props.EnablePurge = utils.Bool(d.Get("purge_enabled").(bool))
+	}
+
+	if d.HasChange("public_ip_type") {
+		publicIPType := clusters.PublicIPType(d.Get("public_ip_type").(string))
+		props.PublicIPType = pointer.To(publicIPType)
+	}
+
+	if d.HasChange("public_network_access_enabled") {
+		publicNetworkAccess := clusters.PublicNetworkAccessEnabled
+		if !d.Get("public_network_access_enabled").(bool) {
+			publicNetworkAccess = clusters.PublicNetworkAccessDisabled
+		}
+		props.PublicNetworkAccess = pointer.To(publicNetworkAccess)
+	}
+
+	if d.HasChange("streaming_ingestion_enabled") {
+		props.EnableStreamingIngest = utils.Bool(d.Get("streaming_ingestion_enabled").(bool))
+	}
+
+	if d.HasChange("trusted_external_tenants") {
+		props.TrustedExternalTenants = expandTrustedExternalTenants(d.Get("trusted_external_tenants").([]interface{}))
+	}
+
+	if d.HasChange("virtual_network_configuration") {
+		if v, ok := d.GetOk("virtual_network_configuration"); ok {
+			if vnetConfig := expandKustoClusterVNET(v.([]interface{})); vnetConfig != nil {
+				props.VirtualNetworkConfiguration = vnetConfig
+			}
+		} else {
+			props.VirtualNetworkConfiguration.State = pointer.To(clusters.VnetStateDisabled)
+			props.PublicNetworkAccess = pointer.To(clusters.PublicNetworkAccessDisabled)
+		}
```

Comment:
I thought `PublicNetworkAccess` and `VirtualNetworkConfiguration.State` were independent of each other and could be toggled seperately?
```suggestion
		} else {
			// 'State' is hardcoded to 'Disabled' for the 'None' pattern.
			// If the vNet block is present it is enabled, if the vNet block is removed it is disabled.
			props.VirtualNetworkConfiguration.State = pointer.To(clusters.VnetStateDisabled)
		}
```

Reply:
Setting the state to false will automatically set public network access to disabled. In this way, users need to manually set public network access property and let me write that down in doc.